### ISSUE TITLE
resolve messages from maven-compiler-plugin

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#2130]
"mvn clean verify" have some messages from compiler that should be resolved or suppressed.

```
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ checkstyle ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 802 source files to checkstyle/target/test-classes
[INFO] checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheckTest.java: Some input files use or override a deprecated API.
[INFO] checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheckTest.java: Recompile with -Xlint:deprecation for details.
[INFO] checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/coding/InputUnnecessaryParentheses.java: Some input files use unchecked or unsafe operations.
[INFO] checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/coding/InputUnnecessaryParentheses.java: Recompile with -Xlint:unchecked for details.
```

```
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ checkstyle ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 802 source files to checkstyle/target/test-classes
[INFO] checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheckTest.java: Some input files use or override a deprecated API.
[INFO] checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractTypeAwareCheckTest.java: Recompile with -Xlint:deprecation for details.
[INFO] checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/coding/InputUnnecessaryParentheses.java: Some input files use unchecked or unsafe operations.
[INFO] checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/coding/InputUnnecessaryParentheses.java: Recompile with -Xlint:unchecked for details.
```

**Comment by mkordas** [comment]
I don't think we can fully resolve that issue. We have warnings like
```
[WARNING] /C:/IdeaProjects/checkstyle/target/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammars/GeneratedJavaLexer.java:[79,21] unchecked call to put(K,V) as a member of the raw type java.util.Hashtable
```
 or
```
[WARNING] /C:/IdeaProjects/checkstyle/target/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammars/GeneratedJavaRecognizer.java:[211,39] redundant cast to antlr.collections.AST
```
 in generated sources or 
```
[WARNING] /C:/IdeaProjects/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/coding/InputFallThrough2.java:[20,13] possible fall-through into case
```
where we test fall-through.

What do you think @romani?

**Comment by mkordas** [comment]
We have also deprecated import: `[WARNING] /C:/IdeaProjects/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java:[39,46] com.puppycrawl.tools.checkstyle.checks.AbstractTypeAwareCheck in com.puppycrawl.tools.checkstyle.checks has been deprecated`

The only solutions I fount at [StackOverflow](http://stackoverflow.com/questions/1858021/suppress-deprecated-import-warning-in-java) are to use star import of fully-qualified class name - but this would trigger PMD and IntelliJ inspection violations that would need to be suppressed. What should I do?

**Comment by romani** [comment]
1)
```
<plugin>
<groupId>org.apache.maven.plugins</groupId>
<artifactId>maven-compiler-plugin</artifactId>
<version>3.3</version>
<configuration>
  <source>${java.version}</source>
  <target>${java.version}</target>
  <compilerArgs>
    <arg>-Xlint:deprecation</arg>
  </compilerArgs>
</configuration>
</plugin>
```

[WARNING] checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java:[39,46] com.puppycrawl.tools.checkstyle.checks.AbstractTypeAwareCheck in com.puppycrawl.tools.checkstyle.checks has been deprecated
[INFO] checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java: Some input files use unchecked or unsafe operations.

Fixed only at Java9: http://bugs.java.com/view_bug.do?bug_id=6598104
The only choice I see is to ignore that message

2)
```
<arg>-Xlint:deprecation</arg>
```

[INFO] Compiling 334 source files to checkstyle/target/classes
[WARNING] checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java:[246,23] 
convert(java.lang.Class,java.lang.Object) in com.puppycrawl.tools.checkstyle.api.AutomaticBean.RelaxedStringArrayConverter 
implements <T>convert(java.lang.Class<T>,java.lang.Object) in org.apache.commons.beanutils.Converter
  return type requires unchecked conversion from java.lang.Object to T
....... a lot of warnings from 
target/generated-sources/antlr/com/puppycrawl/tools/checkstyle/grammars/GeneratedJavaLexer.java

https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AutomaticBean.java#L246

I am not sure how to fix or suppress.

**Comment by mkordas** [comment]
> The only choice I see is to ignore that message
We can also use it as fully qualified name in the code and suppress IDEA/PMD. Would you accept this solution?

**Comment by romani** [comment]
It is easier to ignore that during maven build, i created an issue with hope it is easy to fix, that is even not a warning in logs. Are you ok this this ?

**Comment by romani** [comment]
some changes are released at 6.11, rest of warnings should be ignored.
